docs/DeveloperGuide.adoc

```diff
@@ -295,72 +295,352 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The given tutorial does not exist.
+** 1a1. System shows an error message that the given tutorial does not exist.
++
+Use case ends.
+
+* 1b. The hours of the tutorial class for that week has already been clocked.
+** 1b1. Application shows an error message that the class already has its hours clocked
+for that week
++
+Use case ends.
+
+* 1c. The given input is in an invalid format.
+** 1c1. System shows an error message showing the correct format to add a module.
++
+User case ends.
+
+=== Use case: Adding and viewing assignment score to students
+*Preconditions*
+
+1. User selects the command to add assignment and its' scores to the student.
+2. Given module and tutorial exists in the system.
+
+*MSS*
+
+1. User adds assignment to the tutorial class in the module.
+2. User adds the score for each student for the assignment.
+3. User switches to the student performance window.
+4. System displays results across semester for each student.
+
+*Extensions*
+
+[none]
+
+[none]
+* 2a. One of the students does not exist.
+** 2a1. System shows an error message that the given student does not exist.
+** Steps 2-2a2 are repeated for all invalid students.
++
+Use case resumes at step 2
+
+_{More to be added soon. Stay tuned!}_
 
 [appendix]
 == Non Functional Requirements
 
-.  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+=== Business/Domain Rules:
+. System should be able to handle multiple modules and tutorials
+. Number of module and tutorial slots created should be of reasonable amount (< 10)
+. The application’s functionalities should be easily testable.
 
-_{More to be added}_
+=== Constraints:
+. System should be wholly usable by <<CLI, C.L.I>>. (No <<GUI, G.U.I>> -only functionality)
+. The application should work without requiring an Internet connection.
+
+=== Technical Requirements:
+. Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
+
+=== Performance Requirements:
+. System should be able to hold up to 100 students without a noticeable sluggishness in performance for typical usage.
+
+=== Quality Requirements:
+. A user with above average typing speed for regular English text (i.e. not code, not system admin commands)
+should be able to accomplish most of the tasks faster using commands than using the mouse.
+. The application should work even if the user enters partial commands/data and data in
+different formats, or omits certain command prefixes.
+. Product should be usable by someone who just became a Teaching Assistant.
```

Comment:
You might want to clarify why being a new Teaching Assistant might be a problem.